packages/integrations/partytown/src/index.ts

```diff
@@ -21,6 +21,7 @@ export default function createPlugin(options?: PartytownOptions): AstroIntegrati
 	let partytownSnippetHtml: string;
 	const partytownEntrypoint = resolve('@builder.io/partytown/package.json');
 	const partytownLibDirectory = path.resolve(partytownEntrypoint, '../lib');
+	const SELF_DESTRUCT_ON_VIEW_TRANSITION = `;((d,s)=>(s=d.currentScript,d.addEventListener('astro:before-swap',()=>s?.remove(),{once:true})))(document);`
```

Comment:
The logic seems correct. Is this an IIFE to save on code size?

Reply:
Yes, I started with Typescript, chrunched it a bit and then had fun with it. Mainly it fits the style of the original Partytown snippet.

Reply:
Also I needed a var to save currentScript for later. Always better than storing in window.

Reply:
It's pretty neat!